src/main/java/nextstep/subway/service/PathService.java

```diff
@@ -14,15 +17,34 @@
 @Transactional(readOnly = true)
 public class PathService {
     private final LineRepository lineRepository;
+    private final StationRepository stationRepository;
 
-    public PathService(LineRepository lineRepository) {
+    public PathService(LineRepository lineRepository, StationRepository stationRepository) {
         this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
     }
 
-    public PathResponse getPath(Station source, Station target) {
+    public PathResponse getPath(PathRequest pathRequest) {
+        Station sourceStation = null;
+        Station targetStation = null;
+
+        if (pathRequest.getSource() != null) {
+            sourceStation = getStation(pathRequest.getSource());
+        }
+        if (pathRequest.getTarget() != null) {
+            targetStation = getStation(pathRequest.getTarget());
+        }
```

Comment:
두 값중 하나만 없어도 경로탐색은 수행할 수 없으니 최상단에 유효성 검증을 놓으면 더 간결해질 것 같네요. 

```suggestion
        if (pathRequest.getSource() || pathRequest.getTarget()) {
            throw new IllegalArgumentException("Invalid Path Request - source: [], target: {}");
        }
        
        Station sourceStation = getStation(...);
        Station targetStation = getStation(...);
        
        ...
```